README.md

```diff
@@ -802,9 +794,47 @@ affect the precision of string matching:
   - `exact` has no effect on `regex` or `function` arguments.
   - In most cases using a regex instead of a string gives you more control over
     fuzzy matching and should be preferred over `{ exact: false }`.
-- `trim`: Defaults to `true`; trim leading and trailing whitespace.
+- `normalizer`: An optional function which overrides normalization behavior.
+  See [`Normalization`](#normalization).
+
+### Normalization
+
+Before running any matching logic against text in the DOM, `dom-testing-library`
+automatically normalizes that text. By default, normalization consists of
+trimming whitespace from the start and end of text, and collapsing multiple
+adjacent whitespace characters into a single space.
+
+If you want to prevent that normalization, or provide alternative
+normalization (e.g. to remove Unicode control characters), you can provide a
+`normalizer` function in the options object. This function will be given
+a string and is expected to return a normalized version of that string.
+
+Note: Specifying a value for `normalizer` _replaces_ the built-in normalization, but
+you can call `getDefaultNormalizer` to obtain a built-in normalizer, either
```

Comment:
I'd prefer to not allow people to call this function and instead simply export it:

```
export const normalizer = getDefaultNormalizer({trim: true, collapseWhitespace: true})
```

Reply:
Ok, easy to do, but then somebody wouldn't be able to get hold of, say, a default normalizer that only trims but doesn't collapse whitespace.

If you're happy with that, that's fine, and I'll hide `getDefaultNormalizer` and take all mention of `trim` and `collapseWhitespace` out of the docs.

Reply:
> then somebody wouldn't be able to get hold of, say, a default normalizer that only trims but doesn't collapse whitespace.

I'm thinking that's a pretty minimal use-case, and if it is, they just need to copy/paste this:

```js
function getDefaultNormalizer({trim = true, collapseWhitespace = true} = {}) {
  return text => {
    let normalizedText = text
    normalizedText = trim ? normalizedText.trim() : normalizedText
    normalizedText = collapseWhitespace
      ? normalizedText.replace(/\s+/g, ' ')
      : normalizedText
    return normalizedText
  }
}
```

Considering we're getting simplicity in exchange for a tiny bit of flexibility for 1% of use cases I think that's a win :)

Reply:
I'll also have to remove the example for how to match text without trimming, for instance:
`getByText(node, 'text', {normalizer: getDefaultNormalizer({trim: false})})`

There'd be no easy way to do that without `getDefaultNormalizer`: they'd have to implement their own whitespace-collapsing normalizer.  I don't have much of an opinion on that, but presumably the existing `trim` and `collapseWhitespace` options were added because somebody wanted to be able to turn them off?

Reply:
Ah, that makes sense, considering I want to deprecate `trim` as an option. Ok, let's stick with what you have. I'd like to have @alexkrolick give this a look over too, but I'm 👍

Reply:
I think some of the queries initially had different defaults for trim and collapseWhitespace, and they had different enough effects to keep separate.

I'm on the fence about exporting the factory function, but I definitely don't want to encourage copy-pasting the default implementation - that would be opting-out of future changes.

I think this is good as-is and leaves room to come up with future more streamlined APIs for other use cases.